test/parallel/test-net-connect-buffer.js

```diff
@@ -78,28 +70,4 @@ tcp.listen(0, common.mustCall(function() {
                `Received type ${typeof value}`
     }));
   });
-
-  // Write a string that contains a multi-byte character sequence to test that
-  // `bytesWritten` is incremented with the # of bytes, not # of characters.
-  const a = "L'État, c'est ";
-  const b = 'moi';
-
-  // We're still connecting at this point so the datagram is first pushed onto
-  // the connect queue. Make sure that it's not added to `bytesWritten` again
-  // when the actual write happens.
-  const r = socket.write(a, common.mustCall((er) => {
-    console.error('write cb');
-    assert.ok(connected);
-    assert.strictEqual(socket.bytesWritten, Buffer.from(a + b).length);
-    assert.strictEqual(socket.pending, false);
-  }));
-  socket.on('close', common.mustCall(() => {
-    assert.strictEqual(socket.pending, true);
-  }));
-
-  assert.strictEqual(socket.bytesWritten, Buffer.from(a).length);
-  assert.strictEqual(r, false);
-  socket.end(b);
-
-  assert.strictEqual(socket.readyState, 'opening');
```

Comment:
Why did you split the test in two? What did change?

Reply:
The test expects the socket to be useable after `'error'`, which is not a valid assumption (at least not with `autoDestroy`).